compiler/noirc_evaluator/src/ssa/opt/flatten_cfg.rs

```diff
@@ -683,6 +683,29 @@ impl<'f> Context<'f> {
                     );
                     Instruction::RangeCheck { value, max_bit_size, assert_message }
                 }
+                Instruction::Call { func, mut arguments } => match self.inserter.function.dfg[func]
+                {
+                    Value::Intrinsic(Intrinsic::ToBits(_) | Intrinsic::ToRadix(_)) => {
+                        let field = arguments[0];
+                        arguments.remove(0);
+
+                        let argument_type = self.inserter.function.dfg.type_of_value(field);
+                        let casted_condition = self.insert_instruction(
+                            Instruction::Cast(condition, argument_type),
+                            call_stack.clone(),
+                        );
+                        let field = self.insert_instruction(
+                            Instruction::binary(BinaryOp::Mul, field, casted_condition),
+                            call_stack.clone(),
+                        );
+
+                        arguments.insert(0, field);
```

Comment:
```suggestion
                        let argument_type = self.inserter.function.dfg.type_of_value(field);
                        let casted_condition = self.insert_instruction(
                            Instruction::Cast(condition, argument_type),
                            call_stack.clone(),
                        );
                        let field = self.insert_instruction(
                            Instruction::binary(BinaryOp::Mul, field, casted_condition),
                            call_stack.clone(),
                        );

                        arguments[0] = field;
```
Do we need the remove and insert?

Reply:
Yeahhh, I did think this could be better but was a bit lazy. We'd need to make `arguments` mutable but we could just directly overwrite the first element.

Reply:
`arguments` is already mutable no?